docs/architecture-and-design.md

```diff
@@ -386,6 +389,23 @@ Most [Intents](#intents) are easy to break down into  [Primitives](#primitives),
 
+### Path Manager
+The `Path Manager` is responsible for generating a set of paths that don't collide. It is given a set of path objectives and path planner, and it will generate paths using the given path planner and arbitrate between paths to prevent collisions. One approach is to add extra obstacles to cause the path planner to generate paths that avoid a potential collisions; this is what `VelocityObstaclePathManager` implements. One disadvantage of this approach is that it does not account for when robots will occupy particular positions on the path. (Trajectory planning or use of the Minkowski space are possible ideas to account for time.)
+
+### Path Planner
+The `Path Planner` is responsible for path planning a single robot around a single set of obstacles from a given start to a given destination. Note that the interface of the path planner is intentionally decoupled from the rest of AI, so that it doesn't rely on the representation of [Intents](#intents) or [World](#world).
+
+#### Theta Star Path Planner
+The `Theta Star Path Planner` is a path planner that determines a series of points from the start and destination, such that none of the points are in an obstacle and none of the line segments between adjacent points cross an obstacle, optimizing for the shortest total path length.
+
+To explain how Theta Star (Theta\*) works, it's helpful to briefly explain what A Star (A\*) is. A Star (A\*) uses a grid of points to represent the field and uses graph exploration to connect the start and destination. As it explores each point, it checks against each of the given obstacles to see if that point is inside. A Star (A\*) determines the path length by adding up the number edges along the path from start to destination. At each iteration of graph exploration, A Star (A\*) will (1) update a grid coordinate to the predecessor grid coordinate back to the start of the path, and (2) update the path length to the path length of the predecessor back to the start + one more edge length. By tracing back from the destination via the predecessor of each grid coordinate, A Star (A\*) will connect the start and destination with the shortest path. A Star (A\*) paths are limited by how the graph is set up, namely a grid. Specifically, A Star (A\*) paths change direction at right angles since the graph is a grid. One approach to fixing right-angled paths is smoothing in post-processing, but this smoothed path no longer has guarantees about it being the shortest possible path, since A Star (A\*) optimized for the shortest right-angled path, not the shortest smoothed path.
```

Comment:
Is it more accurate to say "by adding up the edge length" instead of "by adding up the number edges"?